angular-spa-ui/js/search-app/search-srv.js

```diff
@@ -0,0 +1,81 @@
+"use strict";
+
+module.exports = function(app) {
+    
+    app.service('promises', function($q, $http){
+
+        function getAsyncData(method, url) {
+            
+            var deferred = $q.defer();
+            
+            $http({ method: method, url: url })
+                .success(function(data){
+                    data.url = url;
+                    deferred.resolve(data);
+                })
+                .error(function(data,status){
+                    deferred.reject({ data : data, status : status });
```

Comment:
Do not leave space between property name and ":". It should looks next way:
{
    propertyName: 'propertyValue'
}